packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -409,10 +452,73 @@ export class PipelineEditor extends React.Component<
     }
   }
 
+  /*
+   * Checks if there is a path from sourceNode to targetNode in the graph.
+   */
+  nodesConnected(
+    sourceNode: string,
+    targetNode: string,
+    links: any[]
+  ): boolean {
+    if (
+      links.find((value: any, index: number) => {
+        return value.srcNodeId == sourceNode && value.trgNodeId == targetNode;
+      })
+    ) {
+      return true;
+    } else {
+      for (const link of links) {
+        if (
+          link.srcNodeId == sourceNode &&
+          this.nodesConnected(link.trgNodeId, targetNode, links)
+        ) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /*
+   * Validates the new link before adding it.
+   * Returns true if invalid
```

Comment:
You should update this comment block to match the jsdoc format (see requests.tsx in application)

(mostly suggested since you specify both purpose and return (usually jsdocs specific purpose, params, and return))